src/quest/QuestModel.cpp

```diff
@@ -99,39 +137,53 @@ QuestModel* QuestModel::FromJSON(const yi::rapidjson::Value &questJSONObject)
 
 void QuestModel::ActivatePreCondition(CYIString condition)
 {
-    // Find and triggers a pre condition
-    for (CYISharedPtr<Condition> pPreCondition : m_preconditions)
+    if (!IsAvailable())
     {
-        if (pPreCondition->GetCondition() == condition)
+        for (CYISharedPtr<Condition> pPreCondition : m_preconditions)
         {
-            pPreCondition->Activate();
+            if (pPreCondition->GetCondition() == condition)
+            {
+                pPreCondition->Activate();
+            }
+        }
+
+        if (IsAvailable())
+        {
+            QuestManager::QuestAvailableSig(this);
         }
     }
 }
 
 void QuestModel::ActivateCondition(CYIString condition)
 {
-    // Traverse all objectives and attempt to trigger this condition
-    for (YI_INT32 i = 0; i < GetRowCount(); ++i)
+    if (!IsCompleted())
     {
-        CYIAny data(GetItemData(GetIndex(i, 0)));
-
-        if (!data.Empty())
+        for (YI_INT32 i = 0; i < GetRowCount(); ++i)
         {
-            CYISharedPtr<QuestObjectiveModel> pObjective = data.Get<CYISharedPtr<QuestObjectiveModel>>();
+            CYIAny data(GetItemData(GetIndex(i, 0)));
+
+            if (!data.Empty())
+            {
+                CYISharedPtr<QuestObjectiveModel> pObjective = data.Get<CYISharedPtr<QuestObjectiveModel>>();
             
-            pObjective->ActivateCondition(condition);
+                pObjective->ActivateCondition(condition);
+            }
+        }        
+
+        if (IsCompleted())
```

Comment:
Method doing 2 things. It shouldn't activate conditions and send the signal for quest completed.
Maybe fire the signal somewhere else?